src/js/experimental/media-chrome-selectmenu.js

```diff
@@ -229,42 +233,31 @@ class MediaChromeSelectMenu extends globalThis.HTMLElement {
     }
 
     // Get the element that enforces the bounds for the list boxes.
-    const bounds =
-      (this.getAttribute('bounds')
-        ? closestComposedNode(this, `#${this.getAttribute('bounds')}`)
-        : this.parentElement) ?? this;
+    const bounds = getBoundsElement(this);
 
     // Choose .offsetWidth which is not affected by CSS transforms.
     const listboxWidth = this.#listbox.offsetWidth;
     const boundsRect = bounds.getBoundingClientRect();
     const position = -Math.max(buttonRect.x + listboxWidth - boundsRect.right, 0);
 
     this.#listbox.style.left = `${position}px`;
-  }
-
-  #toggleExpanded(closeOnly = false) {
-    this.#expanded = !this.#expanded || closeOnly;
-    this.#button.setAttribute('aria-expanded', this.#expanded);
+    this.#listbox.style.maxHeight = `${boundsRect.height - buttonRect.height}px`;
   }
 
   enable() {
-    this.#button.removeAttribute('disabled');
-    this.#button.addEventListener('click', this.#handleButtonClick);
-    this.#button.addEventListener('keydown', this.#keydownListener);
-    this.#listbox.addEventListener('keydown', this.#keydownListener);
-    this.#toggleExpanded();
-    this.#listbox.addEventListener('change', this.#handleOptionChange);
+    this.#button.toggleAttribute('disabled', false);
+    this.addEventListener('change', this.#handleOptionChange);
+    this.addEventListener('keydown', this.#keydownListener);
+    this.addEventListener('click', this.#clickHandler);
```

Comment:
moving to event delegation here makes it easier and safer I think. 
if the button is removed from the selectmenu, there is no click event on the button anymore